code/__HELPERS/text.dm

```diff
@@ -989,3 +989,27 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 		return word + "ay"
 	//otherwise unmutated
 	return word
+	
+/**
+ * The procedure to check the text of the entered text on ntnrc_client.dm
+ *
+ * This procedure is designed to check the text you type into the chat client. 
+ * It checks for invalid characters and the size of the entered text. 
+ */
+/proc/reject_bad_chattext(text, max_length = 256)
+	var/non_whitespace = FALSE
+	var/char = ""
+	if (length(text) > max_length)
+		return
+	else
+		for(var/i = 1, i <= length(text), i += length(char))
```

Comment:
use `for(var/i in 1 to length(text)`
Alternatively, texts are usually stored as arrays so normally you'd be able to do this too but I'm not 100% sure you can do this in byond, so see if it compiles or not
`for(var/i in text)`

Also, try naming the i variable something more detailed, like index.

Reply:
I think that'll break for unicode chars, at least if you want the char and not each byte. Same with length() https://secure.byond.com/docs/ref/#/{notes}/Unicode

Reply:
It can at least be changed to 
`for(var/i in 1 to length(text)` as a micro optimization

Reply:
it needs to increase by the size of the read char each step right?

Reply:
I see what you mean